# Pemilihan tabel, dan penyingkatan data 

Panggil package pandas 

In [1]:
import pandas as pd

Inputkan data data_penjualan_motor2018.xlsx yang disimpan pada file data dan menampilkan 5 data pertama dengan menggunakan .head(). 

In [2]:
motor = pd.read_excel('data/data_penjualan_motor2018.xlsx')
print(motor.head())

      Bulan  Jenis  Jumlah
0  Januari   Honda  345957
1  Februari  Honda  339152
2     Maret  Honda  384187
3     April  Honda  458499
4       Mei  Honda  436727


Menjumlahkan variabel yang bertipe numerik dalam data penjualan motor 2018 ini yaitu variabel jumlah. 

In [3]:
motor.sum(numeric_only=True)

Jumlah    6382780
dtype: int64

Dari output diatas dapat diketahui jumlah penjualan motor pada tahun 2018 yaitu sebanyak 6382780 unit motor baik dari motor jenis Honda, Yamaha, Kawasaki maupun Suzuki. 

Melihat jumlah penjualan sesuai dengan Bulan.

In [4]:
motor.groupby('Bulan', as_index=False).sum()

Bulan  Jumlah
0     Agustus  567961
1       April  580917
2    Desember  453162
3    Februari  439537
4     Januari  136549
5    Januari   345957
6        Juli  593728
7        Juni  375015
8       Maret  535359
9         Mei  589286
10   November  597347
11    Oktober  610295
12  September  557667

Dari tabel diatas dappat diketahui bahwa jumlah penjualan pada  bulan Agustus sebanyak 567961 unit, Bulan April 580917, dan seterusnya dapat dilihat pada tabel diatas. Perlu diketahui Bulan yang akan mengurutkan sesuai abjad. 

Melihat Jumlah penjualan dengan urutan variabel Bulan kemudian variabel Jenis. 

In [5]:
motor.groupby(['Bulan','Jenis'], as_index=False).sum()

Bulan     Jenis  Jumlah
0     Agustus     Honda  443694
1     Agustus  Kawasaki    7016
2     Agustus    Suzuki    8355
3     Agustus    Yamaha  108896
4       April     Honda  458499
5       April  Kawasaki    2282
6       April    Suzuki    6954
7       April    Yamaha  113182
8    Desember     Honda  325076
9    Desember  Kawasaki    5003
10   Desember    Suzuki    8050
11   Desember    Yamaha  115033
12   Februari     Honda  339152
13   Februari  Kawasaki   10456
14   Februari    Suzuki    4500
15   Februari    Yamaha   85429
16    Januari  Kawasaki    7509
17    Januari    Suzuki    6051
18    Januari    Yamaha  122989
19   Januari      Honda  345957
20       Juli     Honda  450622
21       Juli  Kawasaki    7366
22       Juli    Suzuki    8639
23       Juli    Yamaha  127101
24       Juni     Honda  271206
25       Juni  Kawasaki    2302
26       Juni    Suzuki    5357
27       Juni    Yamaha   96150
28      Maret     Honda  384187
29      Maret  Kawasaki   13969
30      Maret    Suzuki    4077
31      Maret    Yamaha  133126
32        Mei     Honda  436727
33        Mei  Kawasaki    4828
34        Mei    Suzuki    7663
35        Mei    Yamaha  140068
36   November     Honda  440659
37   November  Kawasaki    6003
38   November    Suzuki   10002
39   November    Yamaha  140683
40    Oktober     Honda  456582
41    Oktober  Kawasaki    5212
42    Oktober    Suzuki   10489
43    Oktober    Yamaha  138012
44  September     Honda  406841
45  September  Kawasaki    7036
46  September    Suzuki    9371
47  September    Yamaha  134419

Dari tabel diatas pada Bulan Agustus, Jenis Honda terjual sebanyak 443694 unit, Jenis Kawasaki sebanyak 7016 unit, Jenis Suzuki sebanyak 7016 unit dan Yama sebanyak 108896 Unit. Data selanjutnya dapat dilihat pada tabel diatas. 

Melihat data dengan Jenis Yamaha dan pada bulan Januari

In [6]:
motor[(motor['Jenis'] == 'Yamaha') & (motor['Bulan']=='Januari')]

Bulan   Jenis  Jumlah
12  Januari  Yamaha  122989

dari output diatas pada Bulan Januari dan Jenis Yamaha penjualan motor berjumlah 12298 unit. 

Melihat info data

In [7]:
motor.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 3 columns):
Bulan     48 non-null object
Jenis     48 non-null object
Jumlah    48 non-null int64
dtypes: int64(1), object(2)
memory usage: 1.2+ KB


Pada output diatas diinformasikan bahwa Jumlah semua baris yaitu 48 baris (Lihat RangeIndex) dengan variabel Bulan, Jenis dan Jumlah berjumlah 48 data juga. Berarti data tersebut tidak ada yang kosong. Untuk tipe data yang digunakan yaitu Bulan dan Jenis bertipe objek dan Jumlah bertipe integer. 

Melihat rata-rata penjualan

In [8]:
motor.groupby(['Jenis'], as_index=False).mean()

Jenis         Jumlah
0     Honda  396600.166667
1  Kawasaki    6581.833333
2    Suzuki    7459.000000
3    Yamaha  121257.333333

Rata-rata penjualan Jenis Honda sebesar 396600 unit, Kawasaki sebesar 6582 unit, Suzuki sebanyak 7459 unit dan Yamaha sebanyak 121257. 

Melihat pengurutan Jumlah motor dari besar ke kecil dengan Bulan dan Jenis. 

In [9]:
totals = motor.sort_values('Jumlah', 
                           ascending=False).reset_index(drop=True)
totals

Bulan     Jenis  Jumlah
0       April     Honda  458499
1     Oktober     Honda  456582
2        Juli     Honda  450622
3     Agustus     Honda  443694
4    November     Honda  440659
5         Mei     Honda  436727
6   September     Honda  406841
7       Maret     Honda  384187
8    Januari      Honda  345957
9    Februari     Honda  339152
10   Desember     Honda  325076
11       Juni     Honda  271206
12   November    Yamaha  140683
13        Mei    Yamaha  140068
14    Oktober    Yamaha  138012
15  September    Yamaha  134419
16      Maret    Yamaha  133126
17       Juli    Yamaha  127101
18    Januari    Yamaha  122989
19   Desember    Yamaha  115033
20      April    Yamaha  113182
21    Agustus    Yamaha  108896
22       Juni    Yamaha   96150
23   Februari    Yamaha   85429
24      Maret  Kawasaki   13969
25    Oktober    Suzuki   10489
26   Februari  Kawasaki   10456
27   November    Suzuki   10002
28  September    Suzuki    9371
29       Juli    Suzuki    8639
30    Agustus    Suzuki    8355
31   Desember    Suzuki    8050
32        Mei    Suzuki    7663
33    Januari  Kawasaki    7509
34       Juli  Kawasaki    7366
35  September  Kawasaki    7036
36    Agustus  Kawasaki    7016
37      April    Suzuki    6954
38    Januari    Suzuki    6051
39   November  Kawasaki    6003
40       Juni    Suzuki    5357
41    Oktober  Kawasaki    5212
42   Desember  Kawasaki    5003
43        Mei  Kawasaki    4828
44   Februari    Suzuki    4500
45      Maret    Suzuki    4077
46       Juni  Kawasaki    2302
47      April  Kawasaki    2282

Dari tabel diatas didapatkan jumlah penjualan tertinggi yaitu pada bulan April dengan Jenis Honda dengan 458499 unit, tertinggi kedua yaitu pada bulan Oktober dengan Jenis Honda berjumlah 4586582 unit, dan seterusnya dalat dilihat pada tabel diatas. 

Melihat data untuk Jenis Yamaha saja. 

In [10]:
motor1 = motor[motor['Jenis'] == 'Yamaha'].reset_index(drop=True)
motor1[:5]

Bulan   Jenis  Jumlah
0   Januari  Yamaha  122989
1  Februari  Yamaha   85429
2     Maret  Yamaha  133126
3     April  Yamaha  113182
4       Mei  Yamaha  140068

Membuat function untuk melihat Maximal, mean dan minimal data. 

In [12]:
def get_stats(group):
    return {'min': group.min(), 'max': group.max(), 'mean': group.mean()}
conclusion = motor['Jumlah'].groupby(motor['Jenis']).apply(get_stats).unstack()
conclusion

max           mean       min
Jenis                                      
Honda     458499.0  396600.166667  271206.0
Kawasaki   13969.0    6581.833333    2282.0
Suzuki     10489.0    7459.000000    4077.0
Yamaha    140683.0  121257.333333   85429.0

Pada tabel diatas, pada Jenis motor Honda penjualan terbanyak dengan 458499 unit, dengan rata-rata penjualan 396600 unit dan penjualan terkecil 271206 unit, dan seterusnya bisa dilihat pada tabel diatas. 